blog/_posts/2019-08-28-release-process.md

```diff
@@ -0,0 +1,143 @@
+
+# Minor releases
+
+- Minor releases increment the middle digit of Julia's version number, e.g. going from `1.2.3` to `1.3.0`.
+
+- Minor releases may include bug fixes, new features and "minor changes"—which is the term we're using for technically breaking changes that are sufficiently unlikely to break anyone's code _and_ which do, in fact, do not break things in the package ecosystem as determined by running PkgEval to verify that there isn't any breakage.
+
+- Minor releases are also where significant refactorings of internals go, since we should only be refactoring to the extent that is necessary for fixing bugs in patch releases. This means that if you're relying on some internal Julia stuff that's not public, your code might break in a minor release. This is allowed according to SemVer since the change isn't to a public API—so technically it can break at any time; but we will avoid this in patch releases, so minor releases will be where you have to watch out if you rely on internals somehow.
+
+- Minor releases are branched every **four months**, which means that there are three minor releases per year. The rate is controlled by doing timed features freezes for minor releases: every four months, we announce on discourse that the current development version is about to feature freeze (with about two weeks notice); then on the freeze date, we branch a `release-1.3` branch for the minor release and after that no more feature development is allowed on that branch, from which the release will be tagged. More on that process below.
+
+# Major releases
+
+- Major releases can, according to SemVer, break anything at all. However, realistically we know how we want to write Julia code and that's not going to fundamentally change: most user-level code will stay the same in Julia 2.0 even though we're "allowed" to break things. Breaking things for no reason is not what this is about.
+
+- What a major release does allow, however, is fixing obvious API design mistakes—the kinds of bad, confusing APIs that everyone will be glad to be rid of. It also allows changing low-level things that will break some libraries, but which need to be broken in order to make really fundamental improvements to the language. Changes to Julia's AST representation and how macros work fall into this latter category.
+
+- My favorite example of a breaking change that I want to make in Julia 2.0 is to turn multithreading on by default instead of requiring the user to opt in. Technically this is breaking: if someone wrote thread-unsafe code in 1.x, assuming single-threaded execution by default, if we suddenly turn threading on in Julia 1.x, their code will stop working, breaking our SemVer compatibility promise for non-breaking minor releases. So we need to wait until 2.0 to flip that switch.
```

Comment:
I don't understand why we need to wait until 2.0 to enable multithreading by default. The code will only stop working when it is run under in a multithreading context, but on its own, it works perfectly well. And it is up to the person running that code multithreaded to make sure it is threadsafe.

Reply:
Or wait, did you mean that "enable multithreading by default" means spawning tasks to be non-sticky by default? If that's the case, then yes that would be breaking. I thought you means just running with `nthreads() > 1` by default.

Reply:
I mean setting `JULIA_NUM_THREADS` equal to the number of cores by default.

Reply:
I think I explained pretty clearly why it's a breaking change: code that worked, assuming that there will be only one thread will break if there are suddenly multiple threads.